liquibase-core/src/main/java/liquibase/Liquibase.java

```diff
@@ -2202,40 +2195,30 @@ public void run() throws Exception {
         });
     }
 
+    /**
+     * Calculate the checksum for a given identifier
+     *
+     * @deprecated Use {link {@link CommandScope(String)}.
+     */
     public final CheckSum calculateCheckSum(final String changeSetIdentifier) throws LiquibaseException {
-        if (changeSetIdentifier == null) {
-            throw new LiquibaseException(new IllegalArgumentException("changeSetIdentifier"));
-        }
-        final List<String> parts = StringUtil.splitAndTrim(changeSetIdentifier, "::");
-        if ((parts == null) || (parts.size() < CHANGESET_ID_NUM_PARTS)) {
-            throw new LiquibaseException(
-                    new IllegalArgumentException("Invalid changeSet identifier: " + changeSetIdentifier)
-            );
-        }
-        return this.calculateCheckSum(parts.get(CHANGESET_ID_CHANGELOG_PART),
-                parts.get(CHANGESET_ID_CHANGESET_PART), parts.get(CHANGESET_ID_AUTHOR_PART));
+        CommandResults commandResults = new CommandScope("calculateChecksum")
+                .addArgumentValue(DbUrlConnectionCommandStep.DATABASE_ARG, database)
+                .addArgumentValue(CalculateChecksumCommandStep.CHANGESET_IDENTIFIER_ARG, changeSetIdentifier)
+                .addArgumentValue(CalculateChecksumCommandStep.CHANGELOG_FILE_ARG, this.changeLogFile)
+                .execute();
+        return commandResults.getResult(CalculateChecksumCommandStep.CHECKSUM_RESULT);
+
     }
 
+    /**
+     * Calculates the checksum for the values that form a given identifier
+     *
+     * @deprecated Use {link {@link CommandScope(String)}.
+     */
     public CheckSum calculateCheckSum(final String filename, final String id, final String author)
             throws LiquibaseException {
         LOG.info(String.format("Calculating checksum for changeset %s::%s::%s", filename, id, author));
-        final ChangeLogParameters clParameters = this.getChangeLogParameters();
-        final ResourceAccessor resourceAccessor = this.getResourceAccessor();
-        final DatabaseChangeLog changeLog =
-                ChangeLogParserFactory.getInstance().getParser(
-                        this.changeLogFile, resourceAccessor
-                ).parse(this.changeLogFile, clParameters, resourceAccessor);
-
-        // TODO: validate?
-
-        final ChangeSet changeSet = changeLog.getChangeSet(filename, author, id);
-        if (changeSet == null) {
-            throw new LiquibaseException(
-                    new IllegalArgumentException("No such changeSet: " + filename + "::" + id + "::" + author)
-            );
-        }
-
-        return changeSet.generateCheckSum();
+        return this.calculateCheckSum(String.format("%s::%s::%s", filename, id, author));
```

Comment:
Can you avoid the duplication of the string formatting that occurs in the line above?

Reply:
Yes, thanks for spotting this. In fact I removed the log as it is already executed inside the command step.